### ISSUE TITLE
JavadocType: False negative for unknown tag with no description

### ISSUE THREAD

**Issue by sktpy** [checkstyle/checkstyle#14573]
([Documentation](https://checkstyle.org/checks/javadoc/javadoctype.html))

**config.xml**
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType"/>
  </module>
</module>
```
<br>

**Test.java**
```java
/**
 * @val
 */
class Test {}

```
<br>

**Current CLI:**
```
java -jar checkstyle-10.14.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
<br>

**Expected:**
```
[ERROR] Test.java:2:4: Unknown tag 'val'. [JavadocType]
```
<hr>

**Explanation:**
- `val` is an unknown tag (allowUnknownTags by default is false), and hence should produce a violation.
- Note that this false negative **_only occurs when_** the tag is not accompanied by any description,  violation is produced as expected when the unknown tag is associated with description.

**Attention note from maintainers:**
https://github.com/checkstyle/checkstyle/blob/72cf172972605f79fa8e7ddeea4623529d30f730/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java#L149-L150
this Check is not AST based, so fix might be not trivial. We can accept fix only if it is simple and does not cause more regressions or false positives. Ideally we should  reimplement this Check to be ast based, example: https://github.com/checkstyle/checkstyle/blob/72cf172972605f79fa8e7ddeea4623529d30f730/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java#L110

**Comment by sktpy** [comment]
Example showing that violation is produced as expected if the unknown tag is with the description. (same config is used)

**Test.java**
```java
/**
 * @val well well well, how the turntables
 */
class Test {}
```

**CLI:** `[ERROR] Test.java:2:4: Unknown tag 'val'. [JavadocType]`

**Comment by sktpy** [comment]
If this issue is approved, consider I am on it.

**Comment by nrmancuso** [comment]
@sktpy issue is approved, thanks for noticing this!

**Comment by sktpy** [comment]
Fix is implemented, but I already have 3 PRs open atm, which is the max suggested in contribution guidelines.
Will send the PR for this when any of the below are closed:
- [#14529](https://github.com/checkstyle/checkstyle/pull/14529)
- [#14553](https://github.com/checkstyle/checkstyle/pull/14553)
- [#14603](https://github.com/checkstyle/checkstyle/pull/14603)

Or if any maintainer suggests otherwise, I can send it rn.

**Comment by sktpy** [comment]
Resolution on https://github.com/checkstyle/checkstyle/issues/14618 is required so that I can decide how to proceed with this issue.

**Comment by sktpy** [comment]
Pending until https://github.com/checkstyle/checkstyle/issues/14618#issuecomment-2000436900
